src/messages/search/AuthorPredicate.cpp

```diff
@@ -0,0 +1,23 @@
+#include "messages/search/AuthorPredicate.hpp"
+
+namespace chatterino {
+
+AuthorPredicate::AuthorPredicate(const QStringList &authors)
+    : authors_()
+{
+    // Check if any comma-seperated values were passed and transform those
+    for (const auto &entry : authors)
+    {
+        for (const auto &author : entry.split(',', QString::SkipEmptyParts))
+        {
+            this->authors_ << author;
+        }
+    }
+}
+
+bool AuthorPredicate::appliesTo(const Message &message)
+{
+    return authors_.contains(message.displayName, Qt::CaseInsensitive);
```

Comment:
This should check if it contains either the `displayName` or the `loginName` since users can have localized display names which are different than their login names.

Reply:
Resolved in 97ec3433.